mmengine/runner/log_processor.py

```diff
@@ -339,10 +395,20 @@ def _collect_non_scalars(self, runner, mode: str) -> dict:
         # extract log info and remove prefix to `mode_infos` according to mode.
         for prefix_key, value in infos.items():
             if prefix_key.startswith(mode):
-                key = prefix_key.partition('/')[-1]
+                if self.log_with_hierarchy:
+                    key = prefix_key
+                else:
+                    key = self._remove_prefix(mode, prefix_key)
                 mode_infos[key] = value
         return mode_infos
 
+    def _remove_prefix(self, string: str, prefix: str):
+        """Remove the prefix ``train``, ``val`` and ``test`` of the key."""
+        if string.startswith(prefix):
+            return string[len(prefix) + 1:]
```

Comment:
If the prefix contains `/`, the method should return string[len(prefix):].
If the prefix does not contains `/`, the method should check `string.startswith(f'{prefix}/')`.
I prefer the first way.